src/models.rs

```diff
@@ -939,6 +943,98 @@ pub struct ContextDocsResult {
     pub detections: Vec<DetectionResult>,
 }
 
+/// The request format expected in the /api/v2/text/detect/generated endpoint.
+#[derive(Clone, Debug, Serialize, Deserialize)]
+pub struct ChatDetectionHttpRequest {
+    /// The map of detectors to be used, along with their respective parameters, e.g. thresholds.
+    pub detectors: HashMap<String, DetectorParams>,
+
+    // The list of messages to run detections on.
+    pub messages: Vec<clients::openai::Message>,
+}
+
+impl ChatDetectionHttpRequest {
+    /// Upfront validation of user request
+    pub fn validate(&self) -> Result<(), ValidationError> {
+        // Validate required parameters
+        if self.detectors.is_empty() {
+            return Err(ValidationError::Required("detectors".into()));
+        }
+        if self.messages.is_empty() {
+            return Err(ValidationError::Required("messages".into()));
+        }
+
+        Ok(())
+    }
+
+    /// Validates for the "/api/v1/text/chat" endpoint.
+    pub fn validate_for_text(&self) -> Result<(), ValidationError> {
+        self.validate()?;
+        self.validate_messages()?;
+        validate_detector_params(&self.detectors)?;
+        self.validate_text_detector_params()?;
+
+        Ok(())
+    }
+
+    fn validate_text_detector_params(&self) -> Result<(), ValidationError> {
+        for (model_id, detector_params) in &self.detectors {
+            if let Some(stream) = detector_params.get("stream") {
+                match stream.as_bool() {
+                    Some(stream) => {
+                        if stream {
+                            return Err(ValidationError::Invalid(format!("`stream` parameter specified for model `{model_id}` must not be true")));
+                        } else {
+                            return Ok(());
+                        }
+                    }
```

Comment:
I think we don't need the validation of `stream` here. Since it sort of assumes what detector supports or not. 

If the detector doesn't support any parameter, it would be on the detector implementation to validate it.

Reply:
@gkumbhat I've removed this validation.